### PR TITLE
Added code for Maximum Subarray Sum

### PR DESCRIPTION
**Pull request by jaygala223**
### Describe your change:
Added code for Maximum subarray sum

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: other/maximum_subarray.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+def maxSubArray(nums: list[int]) -> int:
```
[the remark this answers is not in the document]
Comment (jaygala223) [reply other/maximum_subarray.py]: fixed

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/pull/4797#issuecomment-931460531

**Comment by jaygala223** [comment]
Changed the PR name. Thanks for pointing it out @cclauss

**Comment by cclauss** [comment]
We got [lots of subarrays](https://github.com/TheAlgorithms/Python/search?q=subarray), how is this one different, faster, better?

**Review by cclauss — APPROVED** [review]

**Pull request merged**